drivers/vcnl40x0/vcnl40x0.c

```diff
@@ -0,0 +1,169 @@
+    }
+
+    /* Acquire exclusive access */
+    i2c_acquire(dev->params.i2c_dev);
+
+    /* Check sensor ID */
+    uint8_t checkid;
+    i2c_read_reg(dev->params.i2c_dev, dev->params.i2c_addr,
+                 VCNL40X0_REG_PRODUCT_ID, &checkid);
+    if ((checkid & VCNL40X0_PRODUCT_MASK_PRODUCT_ID) != VCNL40X0_PRODUCT_ID) {
+        DEBUG("[Error] Wrong device ID: %d\n", checkid);
+        i2c_release(dev->params.i2c_dev);
+        return -VCNL40X0_ERR_NODEV;
+    }
+
+    _set_led_current(dev);
+    _set_command(dev, VCNL40X0_COMMAND_ALL_DISABLE);
+    _set_proximity_rate(dev);
+    _set_command(dev,
+                 VCNL40X0_COMMAND_PROX_ENABLE | VCNL40X0_COMMAND_AMBI_ENABLE);
+    _set_ambient_config(dev);
+
+    /* Release I2C device */
+    i2c_release(dev->params.i2c_dev);
+
+    return VCNL40X0_OK;
+}
+
+uint16_t vcnl40x0_read_proximity(vcnl40x0_t *dev)
+{
+    i2c_acquire(dev->params.i2c_dev);
+
+    _set_command(dev,
+                 VCNL40X0_COMMAND_PROX_ENABLE | VCNL40X0_COMMAND_PROX_ON_DEMAND);
+
+    while (1) {
+        uint8_t result;
+        _get_command(dev, &result);
+
+        if (result & VCNL40X0_COMMAND_MASK_PROX_DATA_READY) {
+            uint8_t prox_buf[2];
+            i2c_read_regs(dev->params.i2c_dev, dev->params.i2c_addr,
+                          VCNL40X0_REG_PROXIMITY_VALUE, &prox_buf, 2);
+            uint16_t res = (uint16_t)((prox_buf[0] << 8) | prox_buf[1]);
+            DEBUG("Proximity measured: %i\n", res);
+            i2c_release(dev->params.i2c_dev);
+            return res;
+        }
+        xtimer_usleep(1);
+    }
+
+    i2c_release(dev->params.i2c_dev);
+}
+
+uint16_t vcnl40x0_read_ambient_light(vcnl40x0_t *dev)
+{
+    i2c_acquire(dev->params.i2c_dev);
+
+    _set_command(dev,
+                 VCNL40X0_COMMAND_AMBI_ENABLE | VCNL40X0_COMMAND_AMBI_ON_DEMAND);
+
+    while (1) {
+         uint8_t result;
+        _get_command(dev, &result);
+        if (result & VCNL40X0_COMMAND_MASK_AMBI_DATA_READY) {
+            uint8_t ambient_buf[2];
+            i2c_read_regs(dev->params.i2c_dev, dev->params.i2c_addr,
+                          VCNL40X0_REG_AMBIENT_VALUE, &ambient_buf, 2);
+            uint16_t res = (uint16_t)((ambient_buf[0] << 8) | ambient_buf[1]);
+            DEBUG("Proximity measured: %i\n", res);
+            i2c_release(dev->params.i2c_dev);
+            return res;
+        }
+        xtimer_usleep(1);
+    }
+
+    i2c_release(dev->params.i2c_dev);
+}
+
+uint16_t vcnl40x0_read_illuminance(vcnl40x0_t *dev)
+{
+    return vcnl40x0_read_ambient_light(dev) >> 2;
+}
+
+void _set_command(vcnl40x0_t *dev, uint8_t command)
+{
+    i2c_write_reg(dev->params.i2c_dev, dev->params.i2c_addr,
+                  VCNL40X0_REG_COMMAND, command);
+}
+
+void _get_command(vcnl40x0_t *dev, uint8_t *command)
+{
+    i2c_read_reg(dev->params.i2c_dev, dev->params.i2c_addr,
+                 VCNL40X0_REG_COMMAND, &command);
+}
+
+static void _set_led_current(vcnl40x0_t *dev)
```

Comment:
I'd expect this function to have parameter `current` and not using hard-wired `dev->params.led_current`.